src/account/operations/balance_finder.rs

```diff
@@ -70,21 +70,18 @@ impl AccountHandle {
                 }))
                 .await?;
 
+            latest_balance.replace(balance.clone());
+
             // break if we didn't find more balance with the new addresses
-            if balance.total <= latest_balance {
+            if balance.total <= latest_balance.as_ref().map_or(&AccountBalance::default(), |v| v).total {
```

Comment:
Is this correct ? Aren't balance and latest_balance the same because of the replace ?